plugins/module_utils/statuscake.py

```diff
@@ -37,33 +37,37 @@ def fetch(self):
                     logger.debug(f"Fetched data: {test}")
                     return test
 
-    def create(self):
-        if "test_type" not in self.data:
-            self.data["test_type"] = "HTTP"
-        if "check_rate" not in self.data:
-            self.data["check_rate"] = 300
-        self._request("post", self.url, data=self.data)
+    def _create(self):
+        if self.data['state'] == 'present':
+            if "test_type" not in self.data:
+                self.data["test_type"] = "HTTP"
+            if "check_rate" not in self.data:
+                self.data["check_rate"] = 300
+            self._request("post", self.url, data=self.data)
 
-    def update(self, item_id):
+    def _update(self, item_id):
         if "test_type" not in self.data:
             self.data["test_type"] = "HTTP"
         if "check_rate" not in self.data:
             self.data["check_rate"] = 300
         self._request("put", f"{self.url}/{item_id}", data=self.data)
     
-    # def delete(self):
-    #     fetch_data = self.fetch()
-    #     self._request("delete", f"{self.url}/{fetch_data['id']}", data=self.data)
+    def delete(self):
+        fetch_data = self.fetch()
+        if self.data["state"] == "absent" and self.data["name"] == fetch_data["name"]:
+            self._request("delete", f"{self.url}/{fetch_data['id']}", data=self.data)
+            logger.info(f"The test for '{self.data['name']}' was deleted")
 
     def save(self):
         fetch_data = self.fetch()
         logger.debug(f"Does '{self.data['name']}' exists in StatusCake? {bool(fetch_data)}.")
         if not fetch_data:
-            self.create()
-            logger.info(f"A new test for {self.data['name']} was created.")
-        if fetch_data:
-            self.update(fetch_data["id"])
-            logger.info(f"The test for '{fetch_data['name']}' was updated.")
+            if self.data['state'] == 'present':
+                self._create()
+                logger.debug(f"A new test for '{self.data['name']}' was created.")
```

Comment:
```suggestion
                logger.info(f"A new test for '{self.data['name']}' was created.")
```

`info` is helpful here, otherwise no logging output is provided when not using ``--verbose`` when a test is created.